File: backend/app/models/orchestration.py
"""Pydantic models for orchestration API requests and responses."""

from typing import List, Optional

from pydantic import BaseModel


class FallbackChainEntry(BaseModel):
    """A single entry in a fallback chain."""

    backend_type: str
    account_id: Optional[int] = None


class SetFallbackChainRequest(BaseModel):
    """Request body for setting a trigger's fallback chain."""

    entries: List[FallbackChainEntry]


class FallbackChainResponse(BaseModel):
    """Response for a fallback chain query."""

    entity_type: str
    entity_id: str
    entries: list


class AccountHealthResponse(BaseModel):
    """Health status for a single account."""

    account_id: int
    account_name: str
    backend_type: str
    is_rate_limited: bool
    rate_limited_until: Optional[str] = None
    cooldown_remaining_seconds: Optional[int] = None
    total_executions: int = 0
    last_used_at: Optional[str] = None
    is_default: bool = False

File: frontend/src/services/api/types/orchestration.ts
/**
 * Orchestration, fallback chain, and account health types.
 */

export interface AccountHealth {
  account_id: string;
  account_name: string;
  backend_id: string;
  backend_type: string;
  backend_name: string;
  is_rate_limited: boolean;
  rate_limited_until: string | null;
  rate_limit_reason: string | null;
  cooldown_remaining_seconds: number | null;
  total_executions: number;
  last_used_at: string | null;
  is_default: boolean;
  plan: string | null;
}

export interface FallbackChainEntry {
  backend_type: string;
  account_id: string | null;
}

export interface FallbackChain {
  entity_type: string;
  entity_id: string;
  entries: Array<{
    chain_order: number;
    backend_type: string;
    account_id: string | null;
  }>;
}
